FIX: the math functions can not be `use`d

amtoine opened the pull request:
before that commit, one could get two errors when `use`ing the script
```bash
> use maths/math_functions.nu
Error: nu::parser::expected_keyword (link)

  × Expected keyword.
    ╭─[maths/math_functions.nu:38:1]
 38 │ ## constants
 39 │ let pi = 3.1415926535897932
    · ─┬─
    ·  ╰── expected def or export keyword
 40 │ let e  = 2.7182818284590452
    ╰────
```
and
```bash
> use maths/math_functions.nu
Error: nu::parser::unexpected_keyword (link)

  × Statement used in pipeline.
     ╭─[maths/math_functions.nu:154:1]
 154 │
 155 │     for $i in 0..($n_cols - 1) {
     ·     ─┬─
     ·      ╰── not allowed in pipeline
 156 │         ($x | column2 $i) | scale-minmax $a $b | wrap ($name_cols | get $i)
     ╰────
  help: 'for' keyword is not allowed in
        pipeline. Use 'for' by itself,
        outside of a pipeline.
```



this PR fixes them by
- removing the `pi` and `e` definitions with `let`
- changing the problematic `for` to an `each`

amtoine commented:
## additional information
| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.75.1                                   |
| branch             |                                          |
| commit_hash        | 66398fbf77d03a6c7fd231da157f410d0446132e |
| build_os           | linux-x86_64                             |
| build_target       | x86_64-unknown-linux-gnu                 |
| rust_version       | rustc 1.66.1 (90743e729 2023-01-10)      |
| rust_channel       | 1.66.1-x86_64-unknown-linux-gnu          |
| cargo_version      | cargo 1.66.1 (ad779e08b 2023-01-10)      |
| pkg_version        | 0.75.1                                   |
| build_time         | 2023-02-18 11:04:21 +01:00               |
| build_rust_channel | release                                  |
| features           | database, default, trash, which, zip     |
| installed_plugins  | gstat                                    |

fdncred commented:
you could/should probably change the `let`s to `const`s, since they are constants, right?

amtoine commented:
> you could/should probably change the `let`s to `const`s, since they are constants, right?
>
mm not sure it's possible on the latest `main`, is it? :thinking: 

when i revert the [first commit of this PR](https://github.com/nushell/nu_scripts/pull/381/commits/27935219060222bb620fce5bae85b5fb781f1842), i get
```bash
> use maths/math_functions.nu *
Error: nu::parser::expected_keyword (link)

  × Expected keyword.
    ╭─[maths/math_functions.nu:38:1]
 38 │ ## constants
 39 │ let pi = 3.1415926535897932
    · ─┬─
    ·  ╰── expected def or export keyword
 40 │ let e  = 2.7182818284590452
    ╰────
```
so let's try adding an `export` to these
```bash
>_ use maths/math_functions.nu *
Error: nu::parser::parse_mismatch (link)

  × Parse mismatch during operation.
    ╭─[maths/math_functions.nu:38:1]
 38 │ ## constants
 39 │ export let pi = 3.1415926535897932
    ·        ─┬─
    ·         ╰── expected def, def-env, alias, use, or env keyword
 40 │ export let e  = 2.7182818284590452
    ╰────
```

and we get the exact same two errors when replacing `let` by `const` each time :confused: 

you tell me :yum:

fdncred commented:
@kubouch What are we missing here? I thought you could use loose `const` in modules like:
```
const pi = 3.1415926535897932
const e  = 2.7182818284590452
```
This is what I get here
```
use math_functions.nu * 
Error: nu::parser::expected_keyword (link)

  × Expected keyword.
    ╭─[math_functions.nu:38:1]
 38 │ ## constants
 39 │ const pi = 3.1415926535897932
    · ──┬──
    ·   ╰── expected def or export keyword
 40 │ const e  = 2.7182818284590452
    ╰────
```
Maybe all consts have to be inside of defs?

fdncred commented:
But @amtoine, we also don't need these `let`/`const` anymore anyway because we added `math pi` and `math e`. Maybe you already said that and I wasn't catching on?

amtoine commented:
> But @amtoine, we also don't need these `let`/`const` anymore anyway because we added `math pi` and `math e`. Maybe you already said that and I wasn't catching on?
>
ooooh no no, i've removed them separately in my other PR => see #382

wanted to keep these two unrelated changes... unrelated :yum:

amtoine commented:
let me fix the confusion :+1:

amtoine commented:
i'm confused :laughing: 

i've removed `e` and `pi` from `maths/math_functions.nu`, 'cause one can use `math e` and `math pi` instead

do i miss something? :confused:

amtoine commented:
> But @amtoine, we also don't need these `let`/`const` anymore anyway because we added `math pi` and `math e`. Maybe you already said that and I wasn't catching on?
>
so i'd say i've said that yeah :yum:

kubouch commented:
Neither `const` not `let` are supported in modules. I was planning to try it but didn't get to it yet.

fdncred commented:
ok, i think we're good here. we have those constants as math commands.

amtoine commented:
coool :yum:
